pandas/src/interval.pyx

```diff
@@ -101,6 +100,54 @@ cdef class Interval(IntervalMixin):
         end_symbol = ']' if self.closed_right else ')'
         return '%s%s, %s%s' % (start_symbol, self.left, self.right, end_symbol)
 
+    def __add__(self, y):
+
+        if isinstance(y, numbers.Number):
+            return Interval(self.left + y, self.right + y)
+        elif isinstance(y, Interval) and isinstance(self, numbers.Number):
+            # It doesn't make sense to have reverse addition ( i.e. 1 + (1, 2) )
+            # return Interval(y.left + self, y.right + self)
+            raise TypeError("unsupported operand type(s) for +: '%s' and '%s'" %
+                            (type(self).__name__, type(y).__name__))
+        else:
+            raise TypeError("unsupported operand type(s) for +: '%s' and '%s'" %
```

Comment:
you can simply `return NotImplemented` to get Python to raise this default error